game-server/data/handlers/ai/instance/empyreanCrucible/SpectralWarriorAI.java

```diff
@@ -30,7 +30,7 @@ protected void handleAttack(Creature creature) {
 	@Override
 	public void handleHpPhase(int phaseHpPercent) {
 		getPosition().getWorldMapInstance().getInstanceHandler().onChangeStage(StageType.START_STAGE_6_ROUND_5);
-		ThreadPoolManager.getInstance().schedule(this::resurrectAllies, 2000);
+		ThreadPoolManager.getInstance().scheduleAtFixedRate(this::resurrectAllies, 0, 2000);
```

Comment:
This one will lead to a never-ending task.
If you start a repeating task, you will need to make sure to cancel it. This can be done by either using recursion or you need to assign it to a Future<?> and cancel it in handleDied(), handleBackHome() and handleDespawned().
We usually go with the Future<?>, so you can find it in several AIs.